firecrown/cosmosis/run.py

```diff
@@ -185,35 +181,163 @@ def _make_cosmosis_params(cosmosis_config):
+        # FireCrown exposes any scipy distribtion as a prior.
+        # CosmoSIS only exposes three of these right now (plus
+        # a couple of others that scipy doesn't support), but
+        # these are by far the most common.
+
+        # This is a key used by other FireCrown tools
+        if name == 'module':
+            continue
+
+        # The
+        kind = p['kind']
+        loc = p['loc']
+        scale = p['scale']
+        # Flat
+        if kind == 'uniform':
+            upper = loc + scale
+            pr = f'uniform {loc} {upper}'
+        # Exponential, only with loc = 0
+        elif kind == 'expon':
+            # This is not currently in CosmoSIS.  It's not hard to add,
+            # and if there is demand Joe can add it.
+            if loc != 0:
+                raise ValueError("CosmoSIS does not currently support exponential "
+                                 "priors with non-zero 'loc'.  If you need this please "
+                                 "open an issue")
+            pr = f'exp {scale}'
+        # Gaussian.
+        elif kind == 'norm':
+            pr = f'norm {loc} {scale}'
+        else:
+            raise ValueError(f"CosmoSIS does not know how to use the prior kind {kind}")
+        # Put these all in a dictionary
+        P['params', name] = pr
+
+    return cosmosis.Inifile(None, override=P)
+
+
+def _setup(data):
+    # In most cosmosis modules this function
+    # is a bit more complicated!  We are hacking it
+    # a bit here - normally the input to this function
+    # is the configuration information for the module.
+    return data
+
+
+def _execute(block, data):
+    data, ini = data
+    # Calculate the firecrown likelihood as a module
+    # This function, which isn't designed for end users,
+    # is the main connection between cosmosis and firecrown.
+    # CosmoSIS builds the block, and passes it to us here.
+    # The block contains all the sample parameters.
+
+    # Create CCL cosmology
+    ccl_values = {}
+
+    for p in RESERVED_CCL_PARAMS:
+        # First look in the block
+        if block.has_value('params', p):
+            ccl_values[p] = block['params', p]
+        # Then in the ini file, for string params
+        elif ini.has_option('firecrown', p):
+            ccl_values[p] = ini.get('firecrown', p)
+
+    cosmo = get_ccl_cosmology(ccl_values)
+
+    # Put all the parameters in the data dictionary,
+    # both CCL-related and others, like nuisance params.
+    all_params = data['parameters'].keys()
+    for p in all_params:
+        # string parameters are excluded here, and potentially others
+        if block.has_value('params', p):
+            data['parameters'][p] = block['params', p]
+
+    # Currently compute_loglike actually computes the posterior
+    # if priors are included. Prevent that from happening since
+    # CosmoSIS is already handling priors
+    if 'priors' in data:
+        data = data.copy()
+        del data['priors']
```

Comment:
This will leave out priors that are not supported by cosmosis but are by firecrown, leaving people pretty confused as to why cosmosis is not applying their prior.

Reply:
Any non-supported priors will have triggered an error earlier on in the `_make_cosmosis_priors`

Reply:
Ahhhh. Works for me!